File: backend/services/post-rust-actix-web-fdm/src/common/api/invalid_password.rs
use crate::common::utils::error::*;

impl super::super::Password {
    pub(in crate::common) fn password_empty() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "PASSWORD_EMPTY".into(),
                reason: "password is empty".to_string(),
                message: "password is empty".to_string(),
            },
        }
    }
    pub(in crate::common) fn password_too_short() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "PASSWORD_TOO_SHORT".into(),
                reason: "password is too short".to_string(),
                message: "password is too short".to_string(),
            },
        }
    }
    pub(in crate::common) fn password_too_long() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "PASSWORD_TOO_LONG".into(),
                reason: "password is too long".to_string(),
                message: "password is too long".to_string(),
            },
        }
    }
}

File: backend/services/post-kt-vertx-fdm/src/main/kotlin/me/jason5lee/post_kt_vertx_fdm/common/api/InvalidPassword.kt
package me.jason5lee.post_kt_vertx_fdm.common.api

import me.jason5lee.post_ktor_mongo_fdm.common.utils.Err
import me.jason5lee.post_ktor_mongo_fdm.common.utils.FailureBody

internal object InvalidPassword {
    internal val empty: FailureBody = FailureBody(
        error = Err(
            error = "PASSWORD_EMPTY",
            reason = "Password cannot be empty",
            message = "Password cannot be empty",
        )
    )
    internal val tooShort: FailureBody = FailureBody(
        error = Err(
            error = "PASSWORD_TOO_SHORT",
            reason = "Password must be at least 5 characters",
            message = "Password must be at least 5 characters",
        )
    )
    internal val tooLong: FailureBody = FailureBody(
        error = Err(
            error = "PASSWORD_TOO_LONG",
            reason = "Due to the limitation of the bcrypt algorithm, password must be at most 72 characters",
            message = "Password must be at most 72 characters",
        )
    )
}
